gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/auth/GcsDelegationTokens.java

```diff
@@ -238,6 +248,27 @@ public boolean isBoundToDT() {
       return getBoundDT();
     }
 
+    // not bounded tokens, but using shared token instantiation strategy.
+    // in this case we create or reuse token per renewer but share across multiple services.
+    if (this.tokenInstantiationStrategy == DelegationTokenInstantiationStrategy.SHARED) {
+      try {
+        return sharedTokens.computeIfAbsent(
+            Optional.ofNullable(renewer),
+            k -> {
+              logger.atFine().log("Created a shared GCS delegation token");
```

Comment:
Log `renewer` here?

Reply:
I noticed that renewer is not logged in other places of this class. I was wondering if that might be a privacy or security concern?

Reply:
The renewer identifies a token-renewing user (e.g. YARN) but not any private secrets owned by that user. It would be safe to log.